lib/ninja.py

```diff
@@ -57,9 +57,12 @@ class _OutputAccumulator:
 
     @staticmethod
     def get_tty_width():
+        tput_cmd = ["tput", "cols"]
+        if os.getenv("TERM") is None:
+            tput_cmd.extend(["-T", "linux"])
```

Comment:
Just tried this on my machine after looking at the man pages for tput and didn't understand how this would fix the issue. Here's what happened when I ran it on my machine

<img width="437" alt="image" src="https://user-images.githubusercontent.com/68433946/150874060-c756bc29-fc12-4c7b-8374-b0873b328158.png">

Reply:
Right, because `linux` is not in your terminfo database---not surprising if you're on macOS.

(To find the list of terminal types on my system, I ran `infocmp -D` which printed the path to the database (`/usr/share/terminfo` on my system), then I ran `find /usr/share/terminfo`. That's a big list of terminal types but it will certainly vary from one system to another.)

I agree, using a random value for `TERM` doesn't seem robust, unless there's some terminal type that's guaranteed to exist on all systems... I'm not sure that there is.

Reply:
Your system has `TERM` set. The control flow of `get_tty_width()` will not append `-T linux` to the `tput_cmd` variables. The function would return 83 in your case.

Reply:
As I mentioned in the description, the value for `TERM` is not set when running in Linux-based environments in AWS CodeBuild and GitHub actions. Adding the `-T linux` in those cases will result into the error message `tput: No value for $TERM and no -T specified` no longer being printed. A column width of 80 will be used in those cases.

Reply:
I think the easiest solution is to return 80 in the case that `TERM` is not set and forgo changing the "`tput` command" to include `-T linux`.

Also, I tried changing the `stdout` to `subprocess.DEVNULL`, but that didn't work for me.

Reply:
> Right, because `linux` is not in your terminfo database---not surprising if you're on macOS.

In a new zsh terminal:

```bash
> echo $TERM
xterm-256color
> tput cols -T xterm-256-color
80
> tput cols -T linux
80
> unset TERM
> tput cols -T xterm-256-color
tput: No value for $TERM and no -T specified
> tput cols -T linux
tput: No value for $TERM and no -T specified
```

The error is not because `linux` is not in the terminfo database, but because there is no value for `$TERM`.